File: Backend/JobsClassLibrary/Classes/Job.cs
using JobsClassLibrary.Enums;

namespace JobsClassLibrary.Classes
{
    public class Job : JobBase
    {
        public Guid JobID { get; set; }
        public JobStatus Status { get; set; }
        public long CreatedAt { get; set; }
        public long StartedAt { get; set; }
        public long CompletedAt { get; set; }
        public long QueuingTime { get; set; }
        public int Progress { get; set; }

        public Job(JobPriority priority, long queuingTime, Guid jobId)
        {
            JobID = jobId;
            Priority = priority;
            QueuingTime = queuingTime;
            Status = JobStatus.Pending; 
            Progress = 0;
        }

        public Job(){}

        public void UpdateProgress(int progress)
        {
            Progress = progress;
        }

        public void MarkCompleted()
        {
            Status = JobStatus.Completed;
            Progress = 100;
        }

        public void MarkFailed()
        {
            Status = JobStatus.Failed;
            Progress = 0;
        }
    }
}

File: Forntend/src/modals/Job.ts
export enum JobStatus {
  Pending = 0,
  InQueue = 1,
  Running = 2,
  Completed = 3,
  Failed = 4,
  Stopped = 5,
}

export enum JobPriority {
  Regular = 0,
  High = 1,
}

export interface Job {
  jobID: string;
  status: JobStatus;
  createdAt: number;
  startedAt: number;
  completedAt: number;
  progress: number;
  name: string;
  priority: JobPriority;
}

export interface JobProgressUpdate {
  jobID: string;
  status: JobStatus;
  progress: number;
}

export interface CreateJobRequest {
  name: string;
  priority: JobPriority;
}
